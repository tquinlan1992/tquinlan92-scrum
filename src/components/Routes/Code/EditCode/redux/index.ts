import { makeSimpleReducer } from 'tquinlan92-typescript-redux-utils';
import { codeThunkActions } from './thunkActions';

export interface EditCodeState {
    code: string;
    _id: string;
}

export const codeInitialState = {
    code: '',
    _id: ''
};

const { actions: simpleActions, reducer } = makeSimpleReducer<EditCodeState>('Code', codeInitialState);

export const codeActions = {
    ...simpleActions,
    ...codeThunkActions
};

export { reducer as editCodeStateReducer };
