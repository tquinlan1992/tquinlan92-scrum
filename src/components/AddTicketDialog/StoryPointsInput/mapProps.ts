import { AppState } from "@headless/store";
import { addTicketDialogActions } from '../redux';

export const mapStateToProps = ({ addTicket }: AppState) => {
    return {
        storyPoint: addTicket.storyPoint
    };
};

export type Props = ReturnType<typeof mapStateToProps>;

export const mapDispatchToProps = {
    setAddTicketState: addTicketDialogActions.set
};

export type Actions = typeof mapDispatchToProps;