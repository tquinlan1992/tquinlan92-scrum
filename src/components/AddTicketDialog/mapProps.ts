import { AppState } from "@headless/store";
import { addTicketDialogActions } from './redux';

export interface AddTicketDialogOwnProps {
    onRequestClose: () => void;
    open: boolean;
    onSubmit: () => void;
}

export const mapStateToProps = ({ addTicket }: AppState, ownProps: AddTicketDialogOwnProps) => {
    return {
        ...addTicket,
        ...ownProps
    };
};

export type Props = ReturnType<typeof mapStateToProps>; 

export const mapDispatchToProps = {
    addTicket: addTicketDialogActions.addTicket,
    setAddTicketState: addTicketDialogActions.set,
    reset: addTicketDialogActions.reset
};

export type ComponentActions = typeof mapDispatchToProps;