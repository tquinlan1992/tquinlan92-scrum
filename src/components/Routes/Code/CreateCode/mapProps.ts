import { AppState } from "@headless/store";
import { createCodeActions } from "./redux";

export const mapStateToProps = (state: AppState) => {
    return {
        code: state.codeState.createCode.code
    };
};

export type Props = ReturnType<typeof mapStateToProps>;

export const mapDispatchToProps = {
    saveCode: createCodeActions.code
};

export type Actions = typeof mapDispatchToProps;