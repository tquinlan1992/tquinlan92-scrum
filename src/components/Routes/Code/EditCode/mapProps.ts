import { AppState } from "@headless/store";
import { editCodeActions } from './redux'
import { AppStateThunkDispatch } from "@src/utils";

export const mapStateToProps = ({codeState}: AppState) => {
    const { code, _id } = codeState.editCode;
    return {
        code,
        _id
    };
};

export type Props = ReturnType<typeof mapStateToProps>;

export function mapDispatchToProps(dispatch: AppStateThunkDispatch) {
    return {
        loadCode: (id: string) => { dispatch(editCodeActions.loadCode(id)) },
        saveCode: (newCode: string) => { dispatch(editCodeActions.saveCode(newCode)) }
    }
};

export type Actions = ReturnType<typeof mapDispatchToProps>;