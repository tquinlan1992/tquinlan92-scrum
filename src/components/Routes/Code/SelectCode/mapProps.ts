import { AppState } from "@headless/store";
import { selectCodeActions } from "./redux";
import { AppStateThunkDispatch } from "@src/utils";

export const mapStateToProps = (state: AppState) => {
    return {
        options: state.codeState.selectCode.options
    };
};

export type Props = ReturnType<typeof mapStateToProps>;

export function mapDispatchToProps(dispatch: AppStateThunkDispatch) {
    return {
        loadCodeOptions: () => dispatch(selectCodeActions.loadCodeOptions())
    }
};

export type Actions = ReturnType<typeof mapDispatchToProps>;