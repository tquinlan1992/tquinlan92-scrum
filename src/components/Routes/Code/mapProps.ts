import { AppState } from "@headless/store";
import { codeActions } from './redux'

export const mapStateToProps = ({codeState}: AppState) => {
    return {
        code: codeState.code
    };
};

export type Props = ReturnType<typeof mapStateToProps>;

export const mapDispatchToProps = {
    loadCode: codeActions.loadCode,
    saveCode: codeActions.saveCode
};

export type Actions = typeof mapDispatchToProps;