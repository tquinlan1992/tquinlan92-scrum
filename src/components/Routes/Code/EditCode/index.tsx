import * as React from 'react';
import { Props, Actions, mapStateToProps, mapDispatchToProps } from './mapProps';
import { connect, Omit } from "react-redux";
import { getRemoteDb } from "@headless/database/pouch";
import { SelectCodeConnect } from '../SelectCode';
import { Code } from '@components/Code';

interface ActionsNoThunk extends Omit<Actions, 'loadCode' | 'saveCode'> {
    loadCode: () => void;
    saveCode: (newCode: string) => void;
}

export class EditCode extends React.Component<Props & ActionsNoThunk> {

    state = {
        tickets: [],
        console: '',
        evalComponents: []
    }

    componentDidMount() {
        this.props.loadCode();
    }

    onCodeChange(newCode: string) {
        this.props.saveCode(newCode);
    }

    consoleLog(text: string) {
        this.setState({ console: text });
    }

    async run() {
        const db = await getRemoteDb();
        const consoleLog = this.consoleLog.bind(this);
        consoleLog;
        db;
        db.getTickets()
        const evalCode = eval(`(function evalThis(){${this.props.code}})()`);
        let data = [];
        try {
            data = await evalCode.data();
            const evalComponents = evalCode.components(data);
            this.setState({ evalComponents });
        } catch (e) {

        }
        // eval(`(${this.props.code})('created with eval function')`);
    }

    getEvalComponent({ type, value }: { type: string; value: any; }) {
        switch (type) {
            case 'label':
                return <h1>{value}</h1>
            default:
                return null;
        }
    }

    render() {
        const codeProps = {
            code: this.props.code,
            saveCode: this.props.saveCode,
            loadCode: this.props.loadCode
        }
        return (
            <React.Fragment>
                <SelectCodeConnect />
                <Code {...codeProps} />
            </React.Fragment>
        );
    }
}

export const EditCodeConnected = connect(mapStateToProps, mapDispatchToProps)(EditCode);

