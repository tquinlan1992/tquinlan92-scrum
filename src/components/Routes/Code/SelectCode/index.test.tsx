import { SelectCode } from './';
import { shallow } from 'enzyme';
import * as React from 'react';
import { AppState } from '@headless/store';
import { mapStateToProps } from '../EditCode/mapProps';
import { getPartialState } from '@src/utils/testUtils';

describe('test snapshot of SelectCode', () => {
    describe('with no options', () => {
    it('should render', () => {
        const props = {
            loadCodeOptions: jest.fn(),
            options: [
            ]
        };
        const result = shallow(<SelectCode {...props} />);
        expect(result).toMatchSnapshot();
    });
});
describe('with one option', () => {
    it('should render', () => {
        const props = {
            loadCodeOptions: jest.fn(),
            options: [
                {
                    _id: '_id1',
                    name: 'name1'
                }
            ]
        };
        const result = shallow(<SelectCode {...props} />);
        expect(result).toMatchSnapshot();
    });
});
describe('with two options', () => {
    it('should render', () => {
        const props = {
            loadCodeOptions: jest.fn(),
            options: [
                {
                    _id: '_id1',
                    name: 'name1'
                },
                {
                    _id: '_id2',
                    name: 'name2'
                }
            ]
        };
        const result = shallow(<SelectCode {...props} />);
        expect(result).toMatchSnapshot();
    });
});
});