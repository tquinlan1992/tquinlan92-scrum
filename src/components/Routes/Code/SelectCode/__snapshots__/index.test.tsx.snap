// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot of SelectCode with no options should render 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(InputLabel))
    htmlFor="age-simple"
  >
    Age
  </WithStyles(WithFormControlContext(InputLabel))>
  <WithStyles(WithFormControlContext(Select))
    inputProps={
      Object {
        "id": "age-simple",
        "name": "age",
      }
    }
    onChange={[Function]}
    value=""
  />
</WithStyles(FormControl)>
`;

exports[`test snapshot of SelectCode with one option should render 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(InputLabel))
    htmlFor="age-simple"
  >
    Age
  </WithStyles(WithFormControlContext(InputLabel))>
  <WithStyles(WithFormControlContext(Select))
    inputProps={
      Object {
        "id": "age-simple",
        "name": "age",
      }
    }
    onChange={[Function]}
    value="_id1"
  >
    <WithStyles(MenuItem)
      value="_id1"
    >
      _id1
    </WithStyles(MenuItem)>
  </WithStyles(WithFormControlContext(Select))>
</WithStyles(FormControl)>
`;

exports[`test snapshot of SelectCode with two options should render 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(InputLabel))
    htmlFor="age-simple"
  >
    Age
  </WithStyles(WithFormControlContext(InputLabel))>
  <WithStyles(WithFormControlContext(Select))
    inputProps={
      Object {
        "id": "age-simple",
        "name": "age",
      }
    }
    onChange={[Function]}
    value="_id1"
  >
    <WithStyles(MenuItem)
      value="_id1"
    >
      _id1
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      value="_id2"
    >
      _id2
    </WithStyles(MenuItem)>
  </WithStyles(WithFormControlContext(Select))>
</WithStyles(FormControl)>
`;
